TourMap/SignUp/ViewModel/SignUpViewModel.swift

```diff
@@ -9,31 +9,28 @@ import Foundation
 class SignUpViewModel {
   
   let validation = Validation()
-  func validateUserSignUpDetails(name: String, email: String, password: String, confirmPassword: String) -> (isValidUserDetail:Bool, message: String) {
-    let isName = validation.isValidName(name: name)
-    let isEmail = validation.isValidEmail(email: email)
-    let ispassword = validation.isValidPassword(password: password)
-    let isConfirmPassword = validation.isValidPassword(password: confirmPassword)
-    let isEqualPassword = password == confirmPassword
+  
+  func validateUserSignUpDetails(with userData: SignUpData, completionHandler: @escaping(Bool, String) -> Void) {
```

Comment:
Nice one. can you use the result type as the completionHandler signature. so you only need to pass one argument at a time.

func validateUserSignUpDetails(with userData: SignUpData, completionHandler: @escaping (Swift.Result<(Bool, String)>) -> Void). 

this way you can just do completionHandler(.success()) and completion(.failure())

then in the view controller you can then do a switch

Reply:
The reason for this is that you don't need the two result at the same time. your view controller just needs to know when there is a success or when there is failure.